File: VB/E3465/Form1.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports System.Reflection
Imports DevExpress.XtraBars
Imports DevExpress.XtraEditors
Imports DevExpress.XtraReports.Design
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
Imports E3465.WPF

Namespace E3465
    Partial Public Class Form1
        Inherits XtraForm
        Private reportTabControl As ReportTabControl
        Private ReadOnly wpfReportPreviewUserControl As New WpfReportPreviewUserControl()
        Public Sub New()
            InitializeComponent()

            ' Hanlde the DesignerHostLoaded event of a Design Panel, 
            ' to add a new page to the ReportTabControl for showing Document Preview.
            AddHandler xrDesignPanel1.DesignerHostLoaded, AddressOf xrDesignPanel1_DesignerHostLoaded
            xrDesignPanel1.OpenReport(New XtraReport())

        End Sub
        Private Sub xrDesignPanel1_DesignerHostLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
            reportTabControl = CType(e.DesignerHost.GetService(GetType(ReportTabControl)), ReportTabControl)
            Dim logic = reportTabControl.GetProperty(Of TabControlLogic)("Logic")
            Dim bar = logic.GetProperty(Of Bar)("StatusBar")
            AddTabPage(bar)
        End Sub

        Private Sub AddTabPage(ByVal bar As Bar)
            Const Caption As String = "WPF Preview"

            Dim button = New BarButtonItem(bar.Manager, Caption) With {.ButtonStyle = BarButtonStyle.Check, .PaintStyle = BarItemPaintStyle.CaptionGlyph, .GroupIndex = 1, .Glyph = Bitmap.FromStream(System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream("WpfPreview.ico"))}
            bar.InsertItem(bar.ItemLinks(reportTabControl.Pages.Count), button)

            Dim page = New BarButtonTabPage(wpfReportPreviewUserControl, Caption, button, ReportCommand.None)
            reportTabControl.AddPage(wpfReportPreviewUserControl, page)

            ' Handle the WPF Preview BarButtonItem click event
            AddHandler button.ItemClick, AddressOf button_ItemClick
        End Sub

        Private Sub button_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs)
            Const WpfPreviewPageIndex As Integer = 4
            reportTabControl.SelectedIndex = WpfPreviewPageIndex
            If wpfReportPreviewUserControl.Report IsNot Nothing Then
                wpfReportPreviewUserControl.Report.Dispose()
            End If
            Dim report = xrDesignPanel1.Report.Clone()
            report.CreateDocument(True)
            wpfReportPreviewUserControl.Report = report
        End Sub
    End Class
End Namespace

File: CS/WindowsFormsApplication1/Form1.cs
using System.Drawing;
using System.Reflection;
using Common;
using DevExpress.XtraBars;
using DevExpress.XtraEditors;
using DevExpress.XtraReports.Design;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
using WindowsFormsApplication1.WPF;

namespace WindowsFormsApplication1 {
    public partial class Form1 : XtraForm {
        ReportTabControl reportTabControl;
        readonly WpfReportPreviewUserControl wpfReportPreviewUserControl = new WpfReportPreviewUserControl();
        public Form1() {
            InitializeComponent();

            // Hanlde the DesignerHostLoaded event of a Design Panel, 
            // to add a new page to the ReportTabControl for showing Document Preview.
            xrDesignPanel1.DesignerHostLoaded += xrDesignPanel1_DesignerHostLoaded;
            xrDesignPanel1.OpenReport(new XtraReport());

        }
        void xrDesignPanel1_DesignerHostLoaded(object sender, DesignerLoadedEventArgs e) {
            reportTabControl = (ReportTabControl)e.DesignerHost.GetService(typeof(ReportTabControl));
            var logic = reportTabControl.GetProperty<TabControlLogic>("Logic");
            var bar = logic.GetProperty<Bar>("StatusBar");
            AddTabPage(bar);
        }

        private void AddTabPage(Bar bar) {
            const string Caption = "WPF Preview";

            var button = new BarButtonItem(bar.Manager, Caption) {
                ButtonStyle = BarButtonStyle.Check,
                PaintStyle = BarItemPaintStyle.CaptionGlyph,
                GroupIndex = 1,
                Glyph = Bitmap.FromStream(Assembly.GetExecutingAssembly().GetManifestResourceStream("WindowsFormsApplication1.WpfPreview.ico"))
            };
            bar.InsertItem(bar.ItemLinks[reportTabControl.Pages.Count], button);

            var page = new BarButtonTabPage(wpfReportPreviewUserControl, Caption, button, ReportCommand.None);
            reportTabControl.AddPage(wpfReportPreviewUserControl, page);

            // Handle the WPF Preview BarButtonItem click event
            button.ItemClick += button_ItemClick;
        }

        void button_ItemClick(object sender, ItemClickEventArgs e) {
            const int WpfPreviewPageIndex = 4;
            reportTabControl.SelectedIndex = WpfPreviewPageIndex;
            if (wpfReportPreviewUserControl.Report != null)
                wpfReportPreviewUserControl.Report.Dispose();
            var report = xrDesignPanel1.Report.Clone();
            report.CreateDocument(true);
            wpfReportPreviewUserControl.Report = report;
        }
    }
}